enarx-keep/src/backend/probe/x86_64.rs

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::backend::Datum;
+
+use std::arch::x86_64::{CpuidResult, __cpuid_count};
+use std::convert::Into;
+
+pub struct CpuId {
+    pub name: &'static str,
+    pub leaf: u32,
+    pub subl: u32,
+    pub func: fn(CpuidResult) -> (bool, Option<String>),
+}
+
+impl Into<Datum> for &CpuId {
```

Comment:
```suggestion
impl From<&CpuId> for Datum {
```
Always implement these conversions using the `From` trait. There is a `core` blanket implementation that implements `Into` in the opposite direction for all types that implement `From`. So if you implement `From`, you get `Into` for free.

Reply:
I think the blanket trait implementation is only occurs if I also `impl From<Datum> for &CpuId` but since going from CpuId to Datum is so "lossy" I felt only having a 1 direction CpuId -> Datum conversion was sensible.

Reply:
@connorkuehl It doesn't work like that.

Reply:
@npmccallum https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d82fce13eb8ccf617c7c6f8b1dd43ddd

Reply:
I was wrong. I see what you mean now.

Reply:
Addressed this in the latest revision